megamek/src/megamek/client/ui/advancedsearch/AdvancedSearchEquipmentClass.java

```diff
@@ -0,0 +1,99 @@
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megamek.client.ui.advancedsearch;
+
+import java.util.Locale;
+
+enum AdvancedSearchEquipmentClass {
+    EMPTY, AUTOCANNON, RAC, ULTRA, LIGHT, MACHINE_GUN, GAUSS, BALLISTIC, PLASMA, ENERGY, LASER, PULSE, RE_ENGINEERED, PPC, TASER, FLAMER, MISSILE,
+    LRM, MRM, SRM, PHYSICAL, AMS, PRACTICAL_PHYSICAL, INFANTRY_SUIT, PROBE;
+
+    public boolean matches(String name) {
+        if (this == EMPTY) {
+            return true;
+        }
+        name = name.toLowerCase(Locale.ROOT);
+        if (name.contains("ammo")) {
+            return false;
+        }
+        if (this == PHYSICAL) {
+            return name.contains("backhoe") || name.contains("saw") || name.contains("whip") || name.contains("claw") ||
+                name.contains("combine") || name.contains("flail") || name.contains("hatchet") || name.contains("driver") ||
+                name.contains("lance") || name.contains("mace") || name.contains("drill") || name.contains("ram") ||
+                name.contains("blade") || name.contains("cutter") || name.contains("shield") || name.contains("welder") ||
+                name.contains("sword") || name.contains("talons") || name.contains("wrecking");
+        } else if (this == PRACTICAL_PHYSICAL) {
+            return name.contains("claw") || name.contains("flail") || name.contains("hatchet") || name.contains("lance")
+                || name.contains("mace") || name.contains("blade") || name.contains("shield") || name.contains("sword")
+                || name.contains("talons");
+        } else if (this == MISSILE) {
+            return name.contains("lrm") || name.contains("mrm") || name.contains("srm");
+        } else if (this == RE_ENGINEERED) {
+            return name.contains("engineered");
+        } else if (this == ENERGY) {
+            return LASER.matches(name) || PPC.matches(name) || FLAMER.matches(name);
+        } else if (this == MACHINE_GUN) {
+            return (name.contains("mg") || name.contains("machine")) && !name.contains("ammo");
+        } else if (this == BALLISTIC) {
+            return AUTOCANNON.matches(name) || GAUSS.matches(name) || MACHINE_GUN.matches(name);
+        } else if (this == RAC) {
+            return name.contains("rotary");
+        } else if (this == ULTRA) {
+            return name.contains("ultraa");
+        } else if (this == INFANTRY_SUIT) {
+            return (name.contains("suit") || name.contains(" kit") || name.contains(", standard") || name.contains(", concealed")
+                || name.contains("clothing") || name.contains("vest") || name.contains("chainmail") || name.contains("parka"))
+                && !name.contains("ecm") && !name.contains("generic") && !name.contains("suite");
+        } else if (this == AMS) {
+            return name.contains("ams") || name.contains("antimiss");
+        } else if (this == PROBE) {
```

Comment:
Should also include
Electronic Warfare (EW) Equipment
Watchdog Composite Electronic Warfare System (CEWS)
Nova Combined Electronic Warfare System (CEWS)

this is BA specific - Improved Sensors